enclave/src/error.rs

```diff
@@ -0,0 +1,29 @@
+use derive_more::{Display, From};
+use sgx_types::sgx_status_t;
+use crate::rpc;
+
+#[derive(Debug, Display, From)]
+pub enum Error {
+	Rpc(rpc::error::Error),
+	Codec(codec::Error),
```

Comment:
If I understand correctly, this error implement the error types the enclave is returning - atleast that's the case for Rpc and Sgx error. What is the codec error type for?

Reply:
On top level, only `sgx_status_t` is returned. These errors are error types that may occur at whatever somewhere in enclave operations. parity_scale_codec errors is one of them.

Reply:
ok, I see

Reply:
Doing this enables us to use the early return `?` without compromising information. With this pattern, we can remove all the logs in the nested functions and only print the error at the top level.

Reply:
I agree that it will certainly enhance the visual part of the code - but what about debugging? The nested logs were helping me alot to quickly find where the errors originated from. Though I guess with better unit testing this will hopefully not be necessary anymore in the future..

Reply:
Why should it be bad for debugging if the error contains enough details?

Reply:
logs include the information in which file, on which line exactly the error happened (inlcuding allowing me to jump there directly by clicking on it, though I can live without this feature), if you provide similiar distinct error information, I agree that debugging will not be compromised.

Reply:
But I guess this discussion is out of scope for this PR, I marked it as resolved.

Reply:
The lines are not included, I agree. Where necessary, we can still use a more verbose pattern. Although I believe this is another topic. Maybe I should not have written 'remove  _all_ the logs.'

Reply:
Yes, that sounds reasonable. I also think code lines are not necessary as long as the error messages are distinct.. Ctrl+shift+F :)